dipy/reconst/shm.py

```diff
@@ -965,3 +965,65 @@ def sh_to_sf_matrix(sphere, sh_order, basis_type=None, return_inv=True, smooth=0
         return B.T, invB.T
 
     return B.T
+
+def sh_to_ap(coeffs_mtx, normal_factor=0.00001):
+    """ Calculates anisotropic power map with a given SH coeffecient matrix
+
+    References
+    __________
+    .. [1]  Dell’Acqua, F., Lacerda, L., Catani, M., Simmons, A., 2014. 
+            Anisotropic Power Maps: A diffusion contrast to reveal low anisotropy tissues from HARDI data, 
+            in: Proceedings of International Society for Magnetic Resonance in Medicine. Milan, Italy.
+
+    Parameters
+    ----------
+    coeffs_mtx : ndarray
+        A ndarray where the last dimension is the
+        SH coeff estimates for that voxel. 
+    normal_factor: float
+        The value to normalize the ap values. Default is 10^-5.
+
+    Returns
+    -------
+    AP : ndarray
+        The resulting power image.
+    """
+
+    dim = coeffs_mtx.shape[:-1]
+    n_coeffs = coeffs_mtx.shape[-1]
+
+    L = 2    # start at L=2
+    sum_n = 1
+
+    def single_L_ap(coeffs_mtx, L=2, power=2):
+        n_start = 1
+        n_L = 2*L+1
+        for l in xrange(2,L,2):
```

Comment:
To make this Python3 compatible, you will need to add the line: 

   from dipy.utils.six.moves import xrange

At the top of this file.